src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,318 @@
+#!/usr/bin/env python
+
+# Copyright JS Foundation and other contributors, http://js.foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import argparse
+import collections
+import random
+import math
+import subprocess
+
+# Local packages
+import settings
+path = os.path.join(settings.TEMPLATE_DIR)
+sys.path.append(path)
+import validate # Function
+from constants import _constants # Constant values
+from parse_args import parse_args # Argument parser
+from write_file import _write_file
+from util import unsigned32
+
+_GENERATED_FILENAME = '''binary-ops-{NUMBER}.js''' # The file name prefix of the generated tests
+_BINARY_OPERATORS = ["add", "sub", "div", "mul", "mod", "xor", "and", "or", "rightshift", "leftshift",
+"unsignedrightshift"] # The binary operators used in this script
+_ARITHMETIC_OPERATORS = ["add", "sub", "div", "mul", "mod"]  # The arithmetic operators used in this script
+_BITWISE_OPERATORS = ["xor", "and", "or", "rightshift", "leftshift"]  # The bitwise operators used in this script
+_UINT32_OPERATORS = ["unsignedrightshift"]
+
+class GenSequence:
+    def __init__(self, options, calc):
+        num_operands = options.operand_count
+        self.num_operands = num_operands
+        self.operators_list = []
+        self.oper_funcs = {
+              "add": self.add,
+              "sub": self.sub,
+              "mul": self.mul,
+              "div": self.div,
+              "mod": self.mod,
+              "xor": self.bitxor,
+              "and": self.bitand,
+              "or": self.bitor,
+              "leftshift": self.bitleftshift,
+              "rightshift": self.bitrightshift,
+              "unsignedrightshift": self.bitunsignedrightshift
+        }
+
+        for op in options.test_cases:
+            if calc == 1:
```

Comment:
Cannot we use symbolic constants for `calc`? `1,2,3` does not provide too detailed information about what it really does.